ideascube/serveradmin/management/commands/csv2pkg.py

```diff
@@ -23,19 +24,29 @@
 """
 
 
+def ordered_representer(dumper, data):
+    return dumper.represent_mapping('tag:yaml.org,2002:map', data.items())
+
+yaml.add_representer(OrderedDict, ordered_representer, Dumper=yaml.SafeDumper)
+
+
 def generate_catalog_yaml_metadata(path, metadata, _type='zipped-medias'):
+    package_metadata = OrderedDict()
+    for k in ('name', 'description', 'language', 'version', 'url', 'size', 'sha256sum', 'type'):
+        if k == 'version':
+            package_metadata[k] = date.today().isoformat()
+        elif k == 'sha256sum':
+            package_metadata[k] = get_file_sha256(path)
+        elif k == 'size':
+            package_metadata[k] = get_file_size(path)
+        elif k == 'type':
+            package_metadata[k] = _type
+        else:
+            package_metadata[k] = metadata[k]
```

Comment:
That's, huh, a bit baroque.

What's wrong with:

```python
    package_metadata = OrderedDict([
        ('name', metadata['name']),
        ('description', metadata['description']),
        ('language', metadata['language']),
        ('version', date.today().isoformat()),
        ('url', metadata['url']),
        ('size', get_file_size(path)),
        ('sha256sum', get_file_sha256(path)),
        ('type', _type),
    ])
```

It's 3 less lines, no `for` loop full of `if`s.

If it were a complicated structure and we were receiving the list of keys from elsewhere then sure, a more systematic approach like you have would be nicer.

But this is such a simple thing, that your way just looks like needless complexity.